python/tests/test_lowlevel.py

```diff
@@ -619,38 +599,35 @@ def test_num_nodes(self):
                         max_node = node
             assert max_node + 1 == ts.get_num_nodes()
 
-    def verify_dump_equality(self, ts):
-        """
-        Verifies that we can dump a copy of the specified tree sequence
-        to the specified file, and load an identical copy.
-        """
-        with open(self.temp_file, "wb") as f:
-            ts.dump(f)
-        with open(self.temp_file, "rb") as f:
-            ts2 = _tskit.TreeSequence()
-            ts2.load(f)
-        assert ts.get_num_samples() == ts2.get_num_samples()
-        assert ts.get_sequence_length() == ts2.get_sequence_length()
-        assert ts.get_num_mutations() == ts2.get_num_mutations()
-        assert ts.get_num_nodes() == ts2.get_num_nodes()
-        records1 = [ts.get_edge(j) for j in range(ts.get_num_edges())]
-        records2 = [ts2.get_edge(j) for j in range(ts2.get_num_edges())]
-        assert records1 == records2
-        mutations1 = [ts.get_mutation(j) for j in range(ts.get_num_mutations())]
-        mutations2 = [ts2.get_mutation(j) for j in range(ts2.get_num_mutations())]
-        assert mutations1 == mutations2
-        provenances1 = [ts.get_provenance(j) for j in range(ts.get_num_provenances())]
-        provenances2 = [ts2.get_provenance(j) for j in range(ts2.get_num_provenances())]
-        assert provenances1 == provenances2
-
-    def test_dump_equality(self):
+    def test_dump_equality(self, tmp_path):
         for ts in self.get_example_tree_sequences():
             tables = _tskit.TableCollection(sequence_length=ts.get_sequence_length())
             ts.dump_tables(tables)
             tables.compute_mutation_times()
             ts = _tskit.TreeSequence()
             ts.load_tables(tables)
-            self.verify_dump_equality(ts)
+            with open(tmp_path / "temp.trees", "wb") as f:
+                ts.dump(f)
+            with open(tmp_path / "temp.trees", "rb") as f:
+                ts2 = _tskit.TreeSequence()
+                ts2.load(f)
+            assert ts.get_num_samples() == ts2.get_num_samples()
```

Comment:
I did all this stuff long-hand before because we didn't have the table collection equality operator - we could just replace with the standard ``ts.tables == ts2.tables`` now.

Reply:
As this is the low-level ts I had to go with:
```python
            tc = _tskit.TableCollection(ts.get_sequence_length())
            ts.dump_tables(tc)
            tc2 = _tskit.TableCollection(ts2.get_sequence_length())
            ts2.dump_tables(tc2)
            assert tc.equals(tc2)
```